File: bot.py
from decouple import config

import telebot

import os

from scraper import getArticleContent, tts
from url_validator import isValidURL

BOT_TOKEN = config("BOT_TOKEN")
if BOT_TOKEN is None:
    print("Enter Your Bot Token in .env file.")
bot = telebot.TeleBot(BOT_TOKEN)



# Handle '/start' and '/help'
@bot.message_handler(commands=['help', 'start'])
def send_welcome(message):
    bot.reply_to(message, "I am your article reader bot.")

#Handle '/tts'
@bot.message_handler(commands=['tts'])
def textToSpeech(message):
    url = message.text
    modified_url = isValidURL(url)
    
    if modified_url == False:
        bot.reply_to(message, "Provide a website URL.")
        
    else:
        link = modified_url[0]
        print(link)
        bot.reply_to(message, "Ok, Processing...it might take sometime")
        wholeText = getArticleContent(link)
        if wholeText is not None:
            title, content = wholeText
            print("scraping complete")
            result = tts(title, content)
        
            if result == True:

                chat_id = message.chat.id
                

                audio = open("audio.mp3", 'rb')
                #bot.send_audio(chat_id, audio)
                bot.send_audio(chat_id, audio,  performer='Article Reader', title=title)
                print("Sent successfully")
                audio.close()
                if True:
                    os.remove("audio.mp3")
            else:
                bot.reply_to(message, "Sorry, failure converting text to audio")
            
        else:
            bot.reply_to(message, "Sorry, Can't Process this Article")


bot.polling()

File: bot.js
require('dotenv').config()

const gTTS = require("gtts")
const { Bot } = require("grammy");
const { InputFile } = require("grammy");

const isValidURL = require("./utils/url_validator")

const BOT_TOKEN = process.env.BOT_TOKEN;



const bot = new Bot(BOT_TOKEN);

bot.command("start", (ctx) => {
    ctx.reply("Hi, Thanks for starting me :)", {
      reply_to_message_id: ctx.msg.message_id,
    });
  });


bot.command("tts",(ctx)=> {
    const msg = ctx.msg.text;
    //console.log(msg);
    if (msg=="/tts"){
        ctx.reply("Please enter a website url!");
        
    }
    else{
        const url = msg.slice(5);
        console.log(url);
        if (isValidURL(url)== true){
            ctx.reply("Processing... might take sometime", {
                reply_to_message_id: ctx.msg.message_id,
              });


            //HANDLE THE TEXT
                
            const Build = require('newspaperjs').Build;
            const Article = require('newspaperjs').Article;


            Article(url)
            .then(result=>{
                const title = result.title;
                const text = result.text;
                const wholeText = title + ".\n " + text;
                //console.log(wholeText);
                console.log('content scraped');


                const gtts = new gTTS(wholeText, 'en');
                const fileName = "audio" + ".mp3";
                //TTS
                gtts.save(fileName, function () {
                    console.log('Success! TTS...');
                    ctx.replyWithAudio(new InputFile(fileName), {
                    title: title,
                    performer : "Article Reader"
                    });
                    console.log("audio is being sent...");
                    

                });
                })
            .catch(reason=>{
                //return false
                console.log(reason);
                ctx.reply(reason);
                });

        
        }
        
    
    }

})


bot.start();